graphs/possible_bipartition.py

```diff
@@ -1,12 +1,44 @@
 # Can be used for BFS
-from collections import deque 
+from collections import  deque
+# from curses import COLORS 
+
+COLORS = ["blue", "green"]
+
+def dfs(dislikes, current_node, painted_graph, current_color):
+    neighbors = dislikes[current_node]
+    next_color = (current_color + 1) % len(COLORS)
+
+    for neighbor in neighbors:
+        color = painted_graph.get(neighbor)
+
+        if not color:
+            painted_graph[neighbor] = COLORS[next_color]
+            if not dfs(dislikes=dislikes, current_node=neighbor, painted_graph=painted_graph, current_color=next_color):
+                return False
+        elif color != COLORS[next_color]:
+            return False
+    
+    return True
+    
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: o(n)
```

Comment:
⏱ Time complexity is actually O(N + E) here where N is the number of nodes in `dislikes` and E is the number of edges in the graph. This is because depth first search will traverse each node and edge once.